packages/quantic/force-app/main/default/lwc/quanticResultChildren/quanticResultChildren.js

```diff
@@ -0,0 +1,196 @@
+import showThread from '@salesforce/label/c.quantic_ShowThread';
+import {LightningElement, api} from 'lwc';
+// @ts-ignore
+import loadingTemplate from './loading.html';
+// @ts-ignore
+import resultChildrenTemplate from './quanticResultChildren.html';
+
+/** @typedef {import("coveo").FoldedCollection} FoldedCollection */
+/** @typedef {import("coveo").FoldedResult} FoldedResult */
+/** @typedef {import("coveo").FoldedResultList} FoldedResultList */
+/** @typedef {import("coveo").ResultTemplatesManager} ResultTemplatesManager */
+
+/**
+ * The `QuanticResultChildren` component is responsible for displaying the child results of a given result.
+ * This component includes two slots, "before-children" and "after-children", which allow for rendering content before and after the list of children, only when children exist.
+ * @category Result Template
+ * @slot before-children - Slot that allows rendering content before the list of children, only when children exist.
+ * @slot after-children - Slot that allows rendering content after the list of children, only when children exist.
+ * @example
+ * <c-quantic-result-children engine-id={engineId} folded-collection={foldedCollection} template-id="example-template-id" folded-result-list-controller={foldedResultListController} result-templates-manager={resultTemplatesManager}>
+ *   <div slot="before-children">Attached documents</div>
+ * </c-quantic-result-children>
+ */
+export default class QuanticResultChildren extends LightningElement {
+  /**
+   * The ID of the engine instance the component registers to.
+   * @api
+   * @type {string}
+   */
+  @api engineId;
+  /**
+   * The folded collection containing the result and its children.
+   * @api
+   * @type {FoldedCollection}
+   */
+  @api foldedCollection;
+  /**
+   * The ID of the template used to display the child results.
+   * @api
+   * @type {string}
+   */
+  @api templateId;
+  /**
+   * The folded result list controller responsible for executing the actions of the folded collection.
+   * @api
+   * @type {FoldedResultList}
+   */
+  @api foldedResultListController;
+  /**
+   * The template manager from which to get registered custom templates.
+   * @api
+   * @type {ResultTemplatesManager}
+   */
+  @api resultTemplatesManager;
+
+  labels = {
+    loadThread,
+    showThread,
+    hideThread,
+    noDocumentsRelated,
+    noMoreDocumentsRelated,
+  };
+
+  /** @type {boolean} */
+  areAllFoldedResultsLoaded = false;
+  /** @type {boolean} */
+  areFoldedResultsExpanded = false;
+  /** @type {boolean} */
+  foldedResultHasMoreChildren = true;
+  /** @type {Array<FoldedResult>} */
+  firstChildrenPartition;
+
+  get displayedChildren() {
+    if (!this.areAllFoldedResultsLoaded) {
+      return this?.foldedCollection?.children ?? [];
+    }
+    return (
+      (this.areFoldedResultsExpanded
+        ? this?.foldedCollection?.children
+        : this.firstChildrenPartition) ?? []
+    );
```

Comment:
Because the user  can collapse the folded result so we need to be able to comeback to the initial state.
Useful to note that the Headless state will not change when collapsing a foldedResult, that's why I'm keeping the firstChildrenPartition, to be able to display the child results properly when the user collapses the thread.

Useful to note as well that the order of the folded results may change after loading more results. for example if the initial state of folded results is: [A,B, after loading all the collection the folded results can become: [A,C,D,B,E].

Reply:
Okay you make your point for the `firstChildrenPartition` 👍 
It makes sense to keep that original array especially if the order has to be kept somehow.

Then in that case though I would have an array for `displayedFoldedResults` and another array for `allFoldedResults`.

So that when you collapse, you can simply put only the `firstChildrenPartition` inside the `displayedFoldedResults`, or when you expand, you can put all the `allFoldedResults` into the `displayedFoldedResults`.

Reply:
that's exactly how it's working right now x)
I don't have the array `allFoldedResults` just because this data is available directly from the headless state after loading all the folded results, so I just use directly  `this?.foldedCollection?.children` which is coming from the Headless state to display all the folded results.